### ISSUE TITLE
Classifier - CutMix: Regularization Strategy to Train Strong Classifiers

### ISSUE THREAD

**Issue by AlexeyAB** [AlexeyAB/darknet#4419]
**CutMix**: Regularization Strategy to Train Strong Classifiers with Localizable Features
* paper: https://arxiv.org/abs/1905.04899v2
* source: https://github.com/clovaai/CutMix-PyTorch

Use for Classifier training:
* `cutmix=1` - will be used CutMix (36%-91%, 9%-64%) 
* `mixup=1` - will be used MixUp (50%, 50%)

Run training with flag `-show_imgs` to see how images are changed (in separate windows and saved to files `aug_... .jpg`) and how labels are changed (see the console).

`./darknet classifier train cfg/imagenet1k_c.data cfg/efficientnet_b0.cfg -topk -show_imgs`

----

![img1](https://user-images.githubusercontent.com/4096485/69902070-6cd4d400-139a-11ea-9ebd-ffd27f0ef36f.png)

----

Results of ResNet50-Faster-RCNN: https://github.com/clovaai/CutMix-PyTorch/issues/13

Backbone | mAP
-- | --
Baseline | 33.27
Mixup-Pretrained | 34.18
Cutout-Pretrained | 34.28
CutMix-Pretrained | 35.16

CutMix-Pretrained shows a meaningful improvement in MS-COCO detection!

**Comment by AlexeyAB** [comment]
@WongKinYiu

Fixed CutMix: https://github.com/AlexeyAB/darknet/commit/4c26117708469a1136cbb5dc44589f63f0d9c139
removed `boundary-aware constraint`: https://github.com/clovaai/CutMix-PyTorch/issues/14

> No, our code does not sample the cropping region with any boundary-aware constraint.

Use `-show_imgs` flag to see how does it work.
`darknet.exe classifier train cfg/imagenet1k_c.data cfg/efficientnet_b0.cfg -topk -show_imgs`

**Comment by WongKinYiu** [comment]
@AlexeyAB OK, I will re-train the models.

**Comment by WongKinYiu** [comment]
@AlexeyAB `cutmix=1` gets Segmentation fault (core dump) in latest repo.

**Comment by AlexeyAB** [comment]
@WongKinYiu After how many iterations it leads to Segmentation fault?
Did you use only `cutmix=1` or together with `mixup=1 mosaic=1`?

**Comment by WongKinYiu** [comment]
@AlexeyAB 0~2 iterations, use together.

**Comment by AlexeyAB** [comment]
@WongKinYiu 
I tried `cutmix=1` and `cutmix=1 mosaic=1` but I can't reproduce this bug.
* Did you use  the latest version of this repo https://github.com/AlexeyAB/darknet or did you move CutMix/Mosaic data augmentation features to your repo?
* Also check `bad.list` file

**Comment by WongKinYiu** [comment]
I used yesterday's repo and trained a detector.

**Comment by AlexeyAB** [comment]
@WongKinYiu Detector doesn't support CutMix yet: 
I added assert yesterday: https://github.com/AlexeyAB/darknet/commit/efc5478a23a3a3c66d6feefc6d6b485f13503bde#diff-2ceac7e68fdac00b370188285ab286f7R891

**Comment by WongKinYiu** [comment]
@AlexeyAB OK, thank you!

**Comment by WongKinYiu** [comment]
@AlexeyAB 

mixup does not get better performance in my experiments but cutmix does, for example:

| Model | Augmentation | Top-1 | Top-5 |
| :---- | :--------:  | :---: | :---: |
| **CSPResNeXt-50** | Normal | 77.9 | 94.0 |
| **CSPResNeXt-50** | Mixup | 77.2 | 94.0 |
| **CSPResNeXt-50** | Cutmix | 78.0 | 94.3 |
| **CSPResNeXt-50** | Cutmix+Mixup | 77.7 | 94.4 |

**Comment by AlexeyAB** [comment]
@WongKinYiu Thanks! Did you test mosaic and cutmix+mosaic?

**Comment by WongKinYiu** [comment]
@AlexeyAB 

Not yet, I just get some available gpus today.
